src/targets/cooking-fire.js

```diff
@@ -0,0 +1,123 @@
+import { ClickableObject } from "../clickable-object.js";
+import { CONSTANTS } from "../constants/constants.js";
+import { characterData } from "../cookie-io.js";
+import { calcLevel } from "../utilities.js";
+import { Button } from "../ui/button.js";
+import { getItemClass } from "../utilities.js";
+
+export class CookingFire extends ClickableObject {
+    name = "Cooking Fire";
+    examineText = "Hot!";
+    actions = [
+        { text: "Cook", func: "clickTarget" },
+        { text: "Examine", func: "examine" },
+    ];
+
+    validMaterials = new Set();
+
+    constructor(scene) {
+        super();
+        this.scene = scene;
+
+        const cameraWidth = scene.cameras.main.width;
+        const cameraHeight = scene.cameras.main.height;
+        const width = 230;
+        const height = 150;
+        const x = cameraWidth / 2 - width;
+        const y = cameraHeight / 2;
+
+        // Add invisible button for cooking fire
+        this.sprite = new Button(scene, x, y, width, height);
+        this.sprite.on("pointerdown", (pointer) => {
+            if (pointer.rightButtonDown() && !pointer.leftButtonDown()) {
+                this.createRightClickMenu(pointer.x, pointer.y, this.actions);
+            } else {
+                this.clickTarget();
+            }
+        });
+
+        // Potential TODO: do we want to be able to cook already cooked food to make them burnt?
+        this.validMaterials.add("Raw Shrimps");
+        this.validMaterials.add("Raw Anchovies");
```

Comment:
I think these can just be done above in the set initialization above unless there's some reason not to